docs/sources/operations/automatic-stream-sharding.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Automatic Stream Sharding
+menuTitle: Automatic stream sharding
+description: Automatic stream sharding can control issues around the per-stream rate limit
+weight: 110
+---
+
+# Automatic stream sharding
+
+Automatic stream sharding will attempt to keep streams under a `desired_rate` by adding new labels and values to
+existing streams. When properly tuned, this should eliminate issues where log producers are rate limited due to the
+per-stream rate limit.
+
+## When to use automatic stream sharding
+
+Large log streams present several problems for Loki, namely increased and uneven resource usage on Ingesters and
+Distributors. The general recommendation is to explore existing log streams for additional label values that are both
+useful for querying and sufficiently low cardinality. There are many cases, however, where no more labels can
+be extracted, or cardinality for a label is dangerously large. To protect itself from such volume leading to operational failure, Loki implements per-stream rate limits;
+but the result is that some data is lost. The per-stream limit also needs human intervention to change, which is not ideal when log volumes increase and decrease.
+
+Loki uses automatic stream sharding to avoid rate limiting and large streams for any log stream by ensuring it is close
+to a configured `desired_rate`.
+
+## How automatic stream sharding works
+
+Automatic stream sharding works by adding a new label, `__stream_shard__`, to streams and incrementing its value to try
+and keep all streams below a configured `desired_rate`.
+
+The feature adds a new API to Ingesters that reports the size of all existing log streams. Once per second, Distributors
+query the API to get a picture of all stream rates in the system. Distributors use the existing stream-rate data and a
+configured `desired_rate` to determine how many shards a given stream should have. The desired number of new log streams
+are created with the label `__stream_shard__` and logs are divided evenly among the streams.
+
+Because automatic stream sharding is reactive and relies on successive calls to Ingesters, the view of current rates is
+always somewhat behind. As a result, the actual size of sharded streams will always be higher than the `desired_rate`.
+In practice, this is still sufficient to keep log producers from being rate limited by per-stream rate limits.
+
+## Enabling and configuring automatic stream sharding
```

Comment:
nit: I would better document configs in a same way like metrics you did below?

1. List all the configs. explain each one what it means and what is it's default value
2. And small example snippet of the configs

Something like

```
-shard-streams.enbaled - Enables sharded stream either per-tenant or global. Default false
-shard-streams.desired_rate - <what it means>. Default 3M
-shard-streams.logging_enabled - <what it means>. Default false.
```

Reply:
I added a link to the configuration docs for this so the options only have one source of truth. I also added descriptions on the configuration page so people can know things like default values.